app/assets/javascripts/discourse/controllers/topic.js.es6

```diff
@@ -563,6 +574,23 @@ export default Ember.Controller.extend(SelectedPostsCount, BufferedContent, {
     return this.get('selectedPostsUsername') !== undefined;
   }.property('selectedPostsUsername'),
 
+  canMergePosts: function() {
+    // Two or more posts selected
+    if (this.get('selectedPostsCount') < 2) return false;
+    // Must be able to delete all of the posts, this makes it so
+    // that the main post of a topic can't be part of the merged posts
+    const selectedPosts = this.get('selectedPosts');
+    let canDelete = true;
+    selectedPosts.forEach(function(p) {
```

Comment:
You can probably do something like `selectedPosts.some(p => p.get('can_delete'))`